src/scripts/sniperino.ts

```diff
@@ -0,0 +1,58 @@
+// Description:
+//  Beat the sniperino
+//  
+//
+// Commands:
+//  sniperino( me)? - start a new game of sniperino 
+//  snipe( me)? - roll a snipe roll in an attempt to beat the sniperino
+//
+// Author:
+//  Steve Shipsey
+
+import { Robot } from "tsbot";
+
+interface ISniper {
+    name: string;
+    gamesPlayed: number;
+    gamesWon: number;
+    currentSnipe?: number;
+
+    getWinRate(): number;
+    getNewSnipe(): number;
+}
+
+class Sniper implements ISniper {
+    public name: string;
+    public gamesPlayed: number;
+    public gamesWon: number;
+    public currentSnipe: number;
```

Comment:
Unless I'm reading this wrong, in TS class members cannot be optional?

http://stackoverflow.com/questions/36746806/how-to-set-optional-class-parameters-in-angular-2-typescript

src/scripts/sniperino.ts(28,24): error TS1112: A class member cannot be declared optional.